src/screens/Topics/index.js

```diff
@@ -66,8 +65,8 @@ const Topics = () => {
     getCacheTopic();
   }, []);
 
-  const handleSelectedLanguage = React.useCallback(
-    (val) => {
+  const handleTopicChange = React.useCallback(
+    (val, topic, categoryIndex) => {
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The function `handleSelectedLanguage` has been renamed to `handleTopicChange` and now accepts two additional parameters: `topic` and `categoryIndex`. This is a good change as it makes the function more flexible and reusable. However, there's no check for the existence of these new parameters before they are used in the function. It would be safer to add checks for these parameters to prevent potential errors.

```diff
+      (val, topic = {}, categoryIndex = 0) => {
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->